src/main/java/com/minecolonies/coremod/entity/mobs/aitasks/RaiderWalkAI.java

```diff
@@ -64,8 +80,9 @@ private boolean walk()
                 walkToCampFire();
                 return false;
             }
+            raider.setTempEnvDamageImmunity(false);
 
-            if (targetBlock == null || raider.blockPosition().distSqr(targetBlock) < 25 || raider.level.getGameTime() > walkTimer)
+            if (targetBlock == null || raider.level.getGameTime() > walkTimer)
             {
                 targetBlock = raider.getColony().getRaiderManager().getRandomBuilding();
```

Comment:
reset walkInBuildingState to false in here

Reply:
good point